backend/fms_core/template_importer/importers/library_qc.py

```diff
@@ -0,0 +1,104 @@
+
+# {{TEMPLATE PROPERTY NAME : DB PROPERTY NAME}
+TEMPLATE_PROPERTY_MAPPING = {
+    "Measured Volume (uL)": "Library Measured Volume",
+    "Volume Used (uL)": "Library Volume Used",
+    "Concentration (ng/uL)": "Library Concentration",
+    "Quality Instrument": "Library Quality Instrument",
+    "Quality Flag": "Library Quality QC Flag",
+    "Quantity Instrument": "Library Quantity Instrument",
+    "Quantity Flag": "Library Quantity QC Flag",   
+}
+
+class LibraryQCImporter(GenericImporter):
+    SHEETS_INFO = LIBRARY_QC_TEMPLATE['sheets info']
+
+    def __init__(self):
+        super().__init__()
+        self.properties_starting_index = PROPERTIES_STARTING_INDEX
+        self.initialize_data_for_template()
+
+    def initialize_data_for_template(self):
+        #Get protocol for SampleQC
+        protocol = Protocol.objects.get(name='Library Quality Control')
+
+         #Preload data
+        self.preloaded_data = {'process': None, 'process_properties': {}}
+
+        self.preloaded_data['process'] = Process.objects.create(protocol=protocol,
+                                                                comment='Library Quality Control (imported from template)')
+
+        # Preload PropertyType objects for the sample qc in a dictionary for faster access
+        try:
+            self.preloaded_data['process_properties'] = { property.name: {'property_type_obj': property } for property in
+                                                         list(PropertyType.objects.filter(object_id=protocol.id))}
+        except Exception as e:
+            self.base_errors.append(f'Property Type could not be found. {e}')
+
+
+    def import_template_inner(self):
+        libraries_sheet = self.sheets['LibraryQC']
+        for row_id, row_data in enumerate(libraries_sheet.rows):
+
+            process_measurement_properties = self.preloaded_data['process_properties']
+
+             #Populate process properties
+            for i, (key, val) in enumerate(row_data.items()):
+                if key in TEMPLATE_PROPERTY_MAPPING.keys():
+                    process_measurement_properties[TEMPLATE_PROPERTY_MAPPING[key]]['value'] = val
+
+            sample_container = {
+                'container_barcode': row_data['Library Container Barcode'],
+                'container_coord': row_data['Library Container Coord']
+            }
+
+            measures = {
+                'initial_volume': float_to_decimal_and_none(row_data['Initial Volume (uL)']),
+                'measured_volume': float_to_decimal_and_none(row_data['Measured Volume (uL)']),
+                'volume_used': float_to_decimal_and_none(row_data['Volume Used (uL)']),
+                'concentration_nm': float_to_decimal_and_none(row_data['Concentration (nM)']),
+                'concentration_uL' : float_to_decimal_and_none(row_data['Concentration (ng/uL)']),
+                'library_size': float_to_decimal_and_none(row_data['Library size (bp)']).to_integral_value()
```

Comment:
Here i think we have to check this with the lab. The model for library was made before we really had any values to store into the library_size, but if i recall the library_size is an average which mean that even if it actually represent an integral size, it would use decimals. We need to check with Janick or Tony if they have decimals to store (since it's used for conversion we'll take as much precision as possible).

Reply:
Following our discussion with the lab, I will leave this as a decimal with zero decimal digits. I will add a `digits` parameter to the `float_to_decimal_and_none` function so that I can specify zero digits after the decimal.